pkg/runtime/wfengine/backends/actors/workflow_actor.go

```diff
@@ -635,11 +649,24 @@ func (wf *workflowActor) runWorkflow(ctx context.Context, actorID string, remind
 				// Setting executionStatus to failed if workflow has failed/terminated/cancelled
 				executionStatus = diag.StatusFailed
 			}
+			wfExecutionElapsedTime = diag.ElapsedSince(state.workflowStartTime)
 		}
 	}
 	return nil
 }
 
+func (wf *workflowActor) recordWorkflowSchedulingLatency(ctx context.Context, isWorkflowExecutionStartedEvent bool, state *workflowState, scheduledStartTimestamp time.Time, workflowName string, executionStatus string) {
+	if isWorkflowExecutionStartedEvent {
+		state.workflowStartTime = time.Now()
+		// If scheduledStartTimestamp is zero, then scheduling latency is zero
```

Comment:
This case needs to be handled separately. In normal scenarios when scheduledStartTime is not given explicitly, workflow triggers immediately but still there would be some latency due to Reminder creation and invocation. Probably need to take the `startTimeStamp` as the time when reminder was created? thoughts @DeepanshuA @mukundansundar ?

Reply:
In that case, we should have same value for both timeStamp and scheduledStartTime, and it's better to read value from only one property. I think you made some changes in [this PR](https://github.com/microsoft/durabletask-go/pull/60/files) to fix scheduleStartTime issue, we should fix this value there so that, it always has correct value, its better if caller does not check different properties for one value.

Reply:
I agree that when it is scheduled immediately, we should consider all times involved i.e. Reminder creation, invocation - basically anything that accounts for time taken before actual execution could trigger.

Reply:
Sure @DeepanshuA, changes has been made, please have a look.